### ISSUE TITLE
Extend impl bug

### ISSUE THREAD

**Issue by ultrabear** [ultrabear/prefix_array_rs#1]
When Extend is passed two identical K which is not already in the array, the array will insert both items, breaking PrefixArray invariants unexpectedly.

This will not cause UB.

**Comment by ultrabear** [comment]
example:
```rust
let mut p = PrefixArraySet::new();
p.extend(["a", "a"]);
// bug!
assert_eq!(p.len(), 2);
```

**Comment by ultrabear** [comment]
Patched as of 0.2.2
